### PR TITLE
Enable stable sort - when elements are equal sort by their position

### PR DESCRIPTION
**Pull request by u01jmg3**
- See issue #1629
  - Example of problem: https://jsfiddle.net/u01jmg3/hffu9h72/
    - Sort on Item Name column
  - Example of fix: https://jsfiddle.net/u01jmg3/hffu9h72/1/
    - Sort on Item Name column

### CODE REVIEW

**Comment by wenzhixin** [comment]
Have you tested the code in IE browser?

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -837,7 +837,17 @@
         var that = this,
             name = this.options.sortName,
             order = this.options.sortOrder === 'desc' ? -1 : 1,
-            index = $.inArray(this.options.sortName, this.header.fields);
+            index = $.inArray(this.options.sortName, this.header.fields),
+            length = Object.keys(this.data).length;
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: IE do not support this method.

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -846,6 +856,8 @@
                 }
                 var aa = getItemField(a, name),
                     bb = getItemField(b, name),
+                    aa_position = getItemField(a, 'position'),
+                    bb_position = getItemField(b, 'position'),
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: I don't know why here use position field, If haven't use this field? Please provide a working jsfiddle with your changed code, thanks.

File: src/bootstrap-table.js (hunk 1 above)
Comment (u01jmg3) [inline src/bootstrap-table.js]: See 4b4ede7 for IE8 fix

File: src/bootstrap-table.js (hunk 2 above)
Comment (u01jmg3) [inline src/bootstrap-table.js]: Fixed with b6bcb8b

**Comment by wenzhixin** [comment]
in this case, it is a bug https://jsfiddle.net/hffu9h72/2/

**Comment by u01jmg3** [comment]
Anything else I need to change for you to merge?

**Comment by wenzhixin** [comment]
@u01jmg3, my suggestion is to add this feature to an extension like: `bootstrap-table-stable-sort.js`, because add a `position` field to every row is not reasonable, your origin data is: `[{name: 'bootstrap-table'}]` and then changed to `[{name: 'bootstrap-table', position: 0}]`, will let user feel confused.

BTW, the type of `this.data` is array, so you can you use:
```
length = this.data.length;
...
$.each(this.data, function (i, row) {
    if (!row.hasOwnProperty('position')) { // or use _postion instead
        row.position = i;
    }
});
```

What do you think?

**Comment by u01jmg3** [comment]
Including a bug fix as an **optional** extension would not be my choice of action but then this is your library.

Including the original position of each data item is the only way of feasibly overcoming this issue and maintaining a stable sorting algorithm.

I do agree it adds marginal bloating to the data array - would `_postion` make more sense to the user? Perhaps I could look at removing this key post-ordering?

**Comment by u01jmg3** [comment]
Still waiting to find out what you would like to do - I'd like to get this PR closed off

**Comment by djhvscf** [comment]
@wenzhixin @u01jmg3 what about if we use the index of the item in the data array provided instead of add a new property to each item?? Because it can creates a new bugs and that is not the idea

**Comment by wenzhixin** [comment]
How to use the index of the item? @djhvscf

**Comment by wenzhixin** [comment]
Added an option: `sortStable`
https://github.com/wenzhixin/bootstrap-table/commit/44cb312df059184ebb12cb338f87990b454973a0
https://jsfiddle.net/wenyi/hffu9h72/3/
Thanks @u01jmg3

**Comment by fvaletas** [comment]
The fix (sortStable  option) does not work for a custom sorter

**Pull request closed without merging**